### DISCUSSION TITLE
Couple (feature) questions

### DISCUSSION THREAD

**Discussion by vanvanni** [pocketbase/pocketbase#476] [Q&A]
Hi there,

To start off I wanting to say this is a amazing tool with great possibilities and easy setup in comparison to other competitions. Only I have some Ideas and question laying around regarding this framework service. I'm aware that we are not on a v1 release yet. I wonder if there is a roadmap available?

**Questions?**
- **Will we be able to set custom ID fields for the records?** In some cases I would love to maintain UUID's as the ID of a record
- **Is there a possibility to create community based auth providers? Or will there be more in the near future?** Would love to see Steam, Microsoft / Minecraft, Twitch. Wordpress and Auth0 providers for PB

**Features**
- **Edge Function** Think would be a every powerful feature to have to listen to hooks and fire some code. Or be based on API calls. I know we could embed it as framework and do it that way, so it would be up for debate, I guess.

**Comment by ganigeorgiev** [comment]
> I wonder if there is a roadmap available?

The v1.0.0 roadmap is located at https://github.com/orgs/pocketbase/projects/2

---

> Will we be able to set custom ID fields for the records? In some cases I would love to maintain UUID's as the ID of a record

All API services supports user defined IDs, but currently there is a restriction and all ids need to be 15 characters string (this is to avoid collisions with multi-value fields when you have a filter rule like `userIds ~ 'YOUR_ID'`; the restriction will be removed in the future after we add support for `IN` equivalent operator when querying json data).

---

> Is there a possibility to create community based auth providers? Or will there be more in the near future?

There will be more in the future, but currently #376 is with highest priority and all new features are pushed back for awhile.

---

> Edge Function Think would be a every powerful feature to have to listen to hooks and fire some code. 

Users are expected to use PocketBase as framework if they want to run custom code as part of the binary.

There are plans to add support for simple [Collection triggers in #317](https://github.com/pocketbase/pocketbase/issues/317).

_Ideally we will have something like a "Workflow builder" with draggable and configurable blocks, but this requires a lot of time and at least for now it's out of the scope for v1.0.0._
